### PR TITLE
Internal code, infra-bionemo, new sub-package creation tool

### PR DESCRIPTION
**Pull request by malcolmgreaves**
Puts a slimmed-down copy of the internal `infra-bionemo` into the `internal/` directory. This contains the original `license_check.py` program, functionality for creating new bionemo `sub-packages`, and functionality to create new standalone Python projects. New Python projects can either be namespaced (PEP 420) or not (aka "simple"). Adds a new GitHub CI stage for running `pytest` & calculates code coverage in `infra-bionemo`.

New bionemo sub-package creation is exposed as the CLI program `create-bionemo-project`. New namespaced project creation is handled by the CLI tool `create-namespaced-project`. Simple projects are created with `create-py-project`.

Developers **MUST** install this new internal Python project by executing `pip install -e internal/infra-bionemo`.
The top-level bionemo meta package (`pyproject.toml`) has been updated to add `infra-bionemo` to the workspace.

Note that the license check script has moved to `infra-bionemo`. It is now accessible as a standalone CLI program: `license-check`.

### CODE REVIEW

**Review by tshimko-nv — APPROVED** [review]

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: pre-commit
+name: Run pre-commit hooks and Run infra-bionemo tests
```
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply .github/workflows/pre-commit.yml:1]: TODO -- I should move this out to its own GH workflow.

File: internal/infra-bionemo/src/infra_bionemo/new_project/exe/bionemo_subpackage.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import click
+
+from infra_bionemo.new_project.api import check, create_on_filesystem, namespace_py_project_structure
+from infra_bionemo.new_project.utils import ask_yes_or_no
+
+
+__all__: Sequence[str] = ()
+
+
+@click.command(help="Create a new bionemo sub-package project")
```
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply internal/infra-bionemo/src/infra_bionemo/new_project/exe/bionemo_subpackage.py:30]: Probably worth a call out here -- these are internal tools, not sub-packages. This is why the filepath is `internal/...` not `sub-packages/...`. This code isn't intended to be distributed. It's for bionemo devs.

File: internal/infra-bionemo/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# bionemo_infra
+Tools maintained by the BioNeMo Framework Infrastructure team.
```
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply internal/infra-bionemo/README.md:2]: Got it, thanks for the feedback. Will do.

File: internal/infra-bionemo/src/infra_bionemo/new_project/__init__.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply internal/infra-bionemo/src/infra_bionemo/new_project/__init__.py:1]: Internally, in the original `infra-bionemo` project, I do have another command/file that lets one create a non-namespaced package. It's easy enough for me to add that here and it sounds like you want it @jstjohn! I'll put that as a TODO.

**Comment by malcolmgreaves** [comment]
```
$ create-bionemo-project -l sub-packages -p bionemo-supermodel
🔨 Creating sub-packages/bionemo-supermodel
🤔 Do you want to depend on bionemo-llm ? [y/n]
>> n
🔨 Creating new project on file system.
✅ Created bionemo-supermodel and added as a dependency to the bionemo-fw package 🎉

# mgreaves @ mgreaves-mlt in ~/bionemo-framework on git@mg/internal_code x .venv [18:12:40]
$ tree sub-packages/bionemo-supermodel
sub-packages/bionemo-supermodel
├── README.md
├── pyproject.toml
├── src
│   └── bionemo
│       └── supermodel
│           └── __init__.py
└── tests
    └── bionemo
        └── supermodel
            └── test_TODO_bionemo_supermodel.py
```

**Comment by malcolmgreaves** [comment]
```
$ create-py-project -p new-outside-bionemo -l here/
🔨 Creating here/new-outside-bionemo
🔨 Creating new project on file system.
✅ Created new-outside-bionemo in here 🎉

# mgreaves @ mgreaves-mlt in ~/bionemo-framework on git@mg/internal_code x .venv [18:14:31]
$ tree here
here
└── new-outside-bionemo
    ├── README.md
    ├── pyproject.toml
    ├── requirements-dev.txt
    ├── requirements-test.txt
    ├── requirements.txt
    ├── src
    │   └── new_outside_bionemo
    │       └── __init__.py
    └── tests
        └── test_new_outside_bionemo
            └── test_TODO_new_outside_bionemo.py
```

**Comment by malcolmgreaves** [comment]
```
$ create-namespaced-project -n mycompany -m newfeatures -l here2
🔨 Creating namespaced project here2/mycompany-newfeatures
🔨 Creating new namespace Python project on file system.
✅ Created namespaced mycompany-newfeatures in here2 🎉

# mgreaves @ mgreaves-mlt in ~/bionemo-framework on git@mg/internal_code x .venv [18:15:45]
$ tree here2
here2
└── mycompany-newfeatures
    ├── README.md
    ├── pyproject.toml
    ├── requirements-dev.txt
    ├── requirements-test.txt
    ├── requirements.txt
    ├── src
    │   └── mycompany
    │       └── newfeatures
    │           └── __init__.py
    └── tests
        └── test_mycompany
            └── test_newfeatures
                └── test_TODO_mycompany_newfeatures.py
```

**Comment by malcolmgreaves** [comment]
Code coverage for `infra-bionemo` is 85%:
https://github.com/NVIDIA/bionemo-framework/actions/runs/11637032169/job/32409521578?pr=273

```
---------- coverage: platform linux, python 3.10.15-final-0 ----------
Name                                                                                                                       Stmts   Miss  Cover
----------------------------------------------------------------------------------------------------------------------------------------------
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/__init__.py                                 0      0   100%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/license_check.py                          180     57    68%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/new_project/__init__.py                     0      0   100%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/new_project/api.py                         87      0   100%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/new_project/exe/__init__.py                 0      0   100%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/new_project/exe/bionemo_subpackage.py      [55](https://github.com/NVIDIA/bionemo-framework/actions/runs/11637032168/job/32409521571?pr=273#step:6:56)      3    95%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/new_project/exe/namespace.py               27      0   100%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/new_project/exe/simple.py                  23      0   100%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/new_project/templates.py                   32      2    94%
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/infra_bionemo/new_project/utils.py                       15      0   100%
----------------------------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                                        419     62    85%
```

**Comment by malcolmgreaves** [comment]
Tests passed: https://prod.blsm.nvidia.com/bionemo-external-bionemo-fw/job/test_pytest/677/

Merging w/ unrelated update from `main` because we don't have merge queues enabled.

**Pull request merged**